core/decorators/async-setup-decorator.ts

```diff
@@ -5,6 +5,7 @@ import { SETUP } from "./_metadata-keys";
 export function AsyncSetup(target: object,
                            decoratedPropertyKey: string,
                            descriptor?: TypedPropertyDescriptor<() => any>) {
+   descriptor = undefined; // Unused
```

Comment:
Haha boo on the compiler not liking it!

I guess my thoughts are yes, agree having the warning for unused arguments is a good idea. The only exception I'd say is in these two cases where we're implementing an interface rather than creating our own. If we control the interface absolutely rip out unused arguments but I guess I just like to see what's going on with what we'd be provided if that makes sense? thoughts?